predicates/interval/interval.go

```diff
@@ -82,66 +85,71 @@ func (s *spec) Name() string {
 	}
 }
 
-func (s *spec) Create(args []interface{}) (routing.Predicate, error) {
-	switch s.typ {
-	case between:
-		if len(args) != 2 {
-			return nil, predicates.ErrInvalidPredicateParameters
-		}
-	default:
-		if len(args) != 1 {
-			return nil, predicates.ErrInvalidPredicateParameters
-		}
+func (s spec) Create(args []interface{}) (routing.Predicate, error) {
+	p := predicate{typ: s, getTime: time.Now}
+	var loc *time.Location
+	switch {
+	case
+		s == between && len(args) == 3 && parseLocation(args[2], &loc) && parseRFCnz(args[0], &p.begin, loc) && parseRFCnz(args[1], &p.end, loc) && p.begin.Before(p.end),
```

Comment:
I personally like this style of the conditions, but got used to splitting such blocks up into smaller, more basic separate blocks due to review feedbacks received in the past. You can consider to do it here, too. Though, I don't mind how it is now at all.